action/rename.php

```diff
@@ -112,6 +113,48 @@ public function handle_ajax(Doku_Event $event) {
         }
     }
 
+    /**
+     * Handle media renames in media manager
+     *
+     * @param Doku_Event $event
+     * @return void
+     */
+    public function handleAjaxMediaManager(Doku_Event $event)
+    {
+        if ($event->data !== 'plugin_move_rename_mediamanager') return;
+
+        $event->preventDefault();
+        $event->stopPropagation();
+
+        global $INPUT;
+
+        $src = cleanID($INPUT->str('src'));
+        $dst = cleanID($INPUT->str('dst'));
+
+        /** @var helper_plugin_move_op $MoveOperator */
+        $MoveOperator = plugin_load('helper', 'move_op');
+
+        if ($src && $dst) {
+            $response = [];
+            header('Content-Type: application/json');
+
+            if (file_exists(mediaFN($dst))) {
+                $response['error'][] = $this->getLang('errorOverwrite');
+            }
+            if (auth_quickaclcheck($dst) < AUTH_CREATE) {
+                $response['error'][] = $this->getLang('errorPermissions');
+            }
```

Comment:
`moveMedia()`, or rather `checkMedia()`, which is called by `moveMedia`, currently performs the following checks:

* Source file exists.
* Delete right on source file.
* Source same as target.
* Target file doesn't exist.
* Upload right on target.

I don't see why these checks need to duplicated here. In particular, for all of these checks, there is already an error message including translations so this is basically adding duplicate code with duplicate messages. However, this is also missing some checks. Further, checking `AUTH_CREATE` is actually wrong I think, this is for creating pages, for uploading media files you need `AUTH_UPLOAD`.

The same is also true for errors that occur in `moveMedia` itself. In fact, `moveMedia` can even generate messages when the move itself succeeds but the old revisions cannot be moved. Will these messages be displayed to the user? I fear this is currently not the case but I think it would be good.

I think the easiest would be to just recover the generated messages from `$MSG`, see [inc/Ajax.php#L293-L297](https://github.com/dokuwiki/dokuwiki/blob/68253c2bb8d5510451aa80bb7b29f7bc4d790a70/inc/Ajax.php#L293-L297) for an example.

Reply:
Thank you, I have removed the duplication and use the existing checks.